Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3468,16 +3468,19 @@ - (void)uiDatabaseDidUpdate:(NSNotification *)notification
         }
         
         [self updateLastVisibleTimestamp];
-        
-        if (scrollToBottom) {
-            [self scrollToBottomAnimated:shouldAnimateScrollToBottom && shouldAnimateUpdates];
+
+        if (scrollToBottom && shouldAnimateUpdates) {
+            [self scrollToBottomAnimated:shouldAnimateScrollToBottom];
         }
     };
     if (shouldAnimateUpdates) {
         [self.collectionView performBatchUpdates:batchUpdates completion:batchUpdatesCompletion];
     } else {
         [UIView performWithoutAnimation:^{
             [self.collectionView performBatchUpdates:batchUpdates completion:batchUpdatesCompletion];
+            if (scrollToBottom) {
+                [self scrollToBottomAnimated:NO];
```

Comment:
Important fix - if we're _not_ animating, scroll immediately, not in the completion handler.  This makes the "new message" much more snappy.